### PR TITLE
Remove NOS-related code from Pixeltable

### PR DESCRIPTION
**Pull request by aaron-siegel**
There are several good reasons to do this now, beyond code hygiene.

- It simplifies the `BatchedFunction` implementation, which will make batched functions easier to iterate on (which we need to do, cf. PXT-154).
- There were a few tests that relied on NOS, but provided useful coverage independent of NOS. As a byproduct of stripping out NOS, I was able to rework these to use substitutes in place of NOS functions.

### CODE REVIEW

**Comment by aaron-siegel** [comment]
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/165)

File: tests/test_exprs.py [hunk 1]
```diff
@@ -553,36 +553,8 @@ def test_img_members(self, img_tbl) -> None:
...
 
-    def test_img_functions(self, img_tbl) -> None:
-        skip_test_if_not_installed('nos')
-        t = img_tbl
-        from pixeltable.functions.pil.image import resize
-        result = t[t.img.resize((224, 224))].show(0)
-        result = t[resize(t.img, (224, 224))].show(0)
-        result = t[blend(t.img, t.img.rotate(90), 0.5)].show(100)
-        print(result)
-        from pixeltable.functions.nos.image_embedding import openai_clip
```
Comment (mkornacker) [inline tests/test_exprs.py:564]: how about just deleting that and leaving the rest in place?

**Review by aaron-siegel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/165)* status: 0 of 14 files reviewed, 1 unresolved discussion (waiting on @mkornacker and @orm011)

___
*[`tests/test_exprs.py` line 0 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/165#gh-1605618725:-NyGvl48AlSvvbl-l4bD:b9n333j) ([raw file](https://github.com/pixeltable/pixeltable/blob/2ab78f4278acf8bdc511e9f4f94272b172bd68b8/tests/test_exprs.py#L0)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map [this GitHub inline comment thread](https://github.com/pixeltable/pixeltable/pull/165#discussion_r1605618725) to the right spot — sorry!)*

how about just deleting that and leaving the rest in place?
</blockquote></details>

I'd thought about it, but it seemed pointless/redundant since there is a new `test_pil.py` that is far more exhaustive.

Is there something specific in this method that you see that's missing from `test_pil.py`? If so, I think the best course is just to incorporate it into the new `test_pil.py`.

**Review by mkornacker — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/165)* status: 0 of 14 files reviewed, 2 unresolved discussions (waiting on @aaron-siegel and @orm011)

___
*[`tests/test_exprs.py` line 568 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/165#-NyH5nfD5Mpuzdk_y-jj:-NyH5nfD5Mpuzdk_y-jk:beeib1g) ([raw file](https://github.com/pixeltable/pixeltable/blob/16d5ca165322551a0b2e0e22a96b25a662d282d5/tests/test_exprs.py#L568)):*
> ```Python
>         print(result)
>         _ = result._repr_html_()
>         _ = t.img.entropy() > 1
> ```

Everything from here on is about expressions, in particular combining image functions with other exprs that can be evaluated by Postgres. Those cases logically belong here, not in test_pil. Can you delete everything that precedes this and just keep the rest here?

**Review by aaron-siegel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/165)* status: 0 of 14 files reviewed, 2 unresolved discussions (waiting on @mkornacker and @orm011)

___
*[`tests/test_exprs.py` line 568 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/165#-NyH5nfD5Mpuzdk_y-jj:-NyKByIECXXCNurpjwrj:b8rvjfg) ([raw file](https://github.com/pixeltable/pixeltable/blob/16d5ca165322551a0b2e0e22a96b25a662d282d5/tests/test_exprs.py#L568)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

Everything from here on is about expressions, in particular combining image functions with other exprs that can be evaluated by Postgres. Those cases logically belong here, not in test_pil. Can you delete everything that precedes this and just keep the rest here?
</blockquote></details>

Ok, done. I resurrected the test and changed its name to `test_image_expressions`.

**Review by mkornacker — APPROVED** [review]

**Pull request merged**